File: service/tournamentmgmt/src/main/java/dev/wlambertz/rallyon/tournamentmgmt/setup/configuration/internal/tournament/usecase/UpdateDraftUseCase.java
package dev.wlambertz.rallyon.tournamentmgmt.setup.configuration.internal.tournament.usecase;

import dev.wlambertz.rallyon.tournamentmgmt.setup.configuration.api.Tournament;

public interface UpdateDraftUseCase {

    Tournament execute(long tournamentId, Tournament draftChanges, long version, long actingUserId);
}

File: service/tournamentmgmt/src/main/kotlin/dev/wlambertz/rallyon/tournamentmgmt/setup/configuration/internal/tournament/usecase/UpdateDraftUseCase.kt
package dev.wlambertz.rallyon.tournamentmgmt.setup.configuration.internal.tournament.usecase

import dev.wlambertz.rallyon.tournamentmgmt.setup.configuration.api.Tournament

interface UpdateDraftUseCase {

    fun execute(tournamentId: Long, draftChanges: Tournament?, version: Long, actingUserId: Long): Tournament
}
